tracing/flags.go

```diff
@@ -0,0 +1,110 @@
+package tracing
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/els0r/telemetry/tracing/internal/flagutil"
+	flag "github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+const (
+	otelGRPC = "otel-grpc"
+	stdout   = "stdout"
+)
+
+var supportedCollectors = []string{
+	otelGRPC,
+	stdout,
+}
+
+// Tracing flag names and their help text. The '.' is the hierarchy delimiter
+const (
+	TracingKey = "tracing"
+
+	TracingEnabledArg     string = TracingKey + ".enabled"
+	TracingEnabledDefault bool   = true
+	TracingEnabledHelp    string = "enable tracing"
+
+	TracingCollectorKey string = TracingKey + ".collector"
+
+	TracingCollectorTypeArg     string = TracingCollectorKey + ".type"
+	TracingCollectorTypeDefault string = ""
+	TracingCollectorTypeHelp    string = "tracing collector type"
+
+	TracingCollectorEndpointArg     string = TracingCollectorKey + ".endpoint"
+	TracingCollectorEndpointDefault string = ""
+	TracingCollectorEndpointHelp    string = "endpoint collecting traces"
+)
+
+type traceFlagsConfig struct {
+	enabled           bool
+	collectorType     string
+	collectorEndpoint string
+}
+
+func (t traceFlagsConfig) options(ctx context.Context) (opts []Option) {
+	if !(t.enabled) {
+		return opts
+	}
+
+	switch t.collectorType {
+	case stdout:
+		opts = append(opts, WithStdoutTraceExporter(true))
+	case otelGRPC:
+		opts = append(opts, WithGRPCExporter(ctx, traceFlags.collectorEndpoint))
+	default:
+		return opts
+	}
+	return opts
+}
+
+var traceFlags traceFlagsConfig
+
+var registered bool
+var registration = sync.Once{}
+
+// RegisterFlags registers CLI arguments into flags. It accepts the Flagger interface, allowing for the use of pflag
+// or other flag libraries such as cobra. Registration of the flags is done exactly once
+func RegisterFlags(flags *flag.FlagSet) {
+	registration.Do(func() {
+		registered = true
+
+		flags.Bool(TracingEnabledArg, TracingEnabledDefault, TracingEnabledHelp)
```

Comment:
I couldn't find any place below where the enabled flag is used. Am I missing something? Similar to metrics etc. tracing should IMHO add as little overhead as possible when _not_ enabled, so I guess there should be some directives somewhere that skip doing stuff when this is disabled.